File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        if (n == 1) return 1;
        std::vector<int> memo(n + 1);
        memo[0] = 0, memo[1] = 1, memo[2] = 2;
        for (int i = 3; i <= n; i++) {
            memo[i] = memo[i - 2] + memo[i - 1];
        }
        return memo[n];
    }
};

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        dp = dict()
        
        def ways(i):
            if i == n:
                return 1
            if i > n:
                return 0
            if i in dp:
                return dp[i]
            dp[i] = ways(i + 1) + ways(i + 2)
            return dp[i]
        
        return ways(0)